src/OidcProxy.Net.Auth0/Auth0IdentityProvider.cs

```diff
@@ -37,24 +37,27 @@ protected override Task<Uri> BuildEndSessionUri(string? idToken, string redirect
     {
         var queryStringSeparator = "?";
 
+        var redirectEndpoint = config.UseOidcLogoutEndpoint ? "/oidc/logout" : "/v2/logout";
+        var redirectQueryParameter = config.UseOidcLogoutEndpoint ? "post_logout_redirect_uri" : "returnTo";
+        
         // Docs on redirect URL
         // https://auth0.com/docs/authenticate/login/logout/redirect-users-after-logout
         var redirectUrl = string.Empty;
         if (!string.IsNullOrEmpty(redirectUri))
         {
             var returnToQueryStringValue = HttpUtility.UrlEncode(redirectUri);
-            redirectUrl = $"{queryStringSeparator}returnTo={returnToQueryStringValue}";
+            redirectUrl = $"{queryStringSeparator}{redirectQueryParameter}={returnToQueryStringValue}";
 
             queryStringSeparator = "&";
         }
         
         // Docs on federated logout:
         // https://auth0.com/docs/authenticate/login/logout/log-users-out-of-idps
         var federated = config.FederatedLogout ? $"{queryStringSeparator}federated" : string.Empty;
-        
+
         // Auth0 does not define their end_session_endpoint in the well-known/openid-configuration
         var query = $"{redirectUrl}{federated}";
-        var endSessionUrl = $"https://{config.Domain}/oidc/logout{query}";
+        var endSessionUrl = $"https://{config.Domain}{redirectEndpoint}{query}";
```

Comment:
Question:

According to this [document](https://auth0.com/docs/authenticate/login/logout/log-users-out-of-auth0) , the url format for signing out  using the `/oidc/logout` endpoint is recommended to be in the following format:

`https://{yourDomain}/oidc/logout?id_token_hint={yourIdToken}&post_logout_redirect_uri={yourCallbackUrl}`

Hence my question: Any particular reason we did not include the `id_token_hint` in the url?

Reply:
Oddly, I thought about that yesterday, but, literally, asked "how do i get the `id_token`.  And since it is only "recommended"  and not "required," I skipped it.

Then I read your comment and looked at the code again, and realized that the id token is passed as a parameter....  This is what I get for trying to code on Sunday.  

PR updated with ID token as a parameter